GridTariffAPI.lib/Services/V2/TariffQueryService.cs

```diff
@@ -29,7 +38,88 @@ public class TariffQueryService : ITariffQueryService
             _serviceHelper = serviceHelper;
         }
 
-        public async Task<GridTariffCollection> QueryTariffAsync(
+        public async Task<TariffQueryRequestMeteringPointsResult> QueryMeteringPointsTariffsAsync(
+            DateTimeOffset paramFromDate,
+            DateTimeOffset paramToDate,
+            List<String> meteringPoints)
+        {
+            var retVal = new TariffQueryRequestMeteringPointsResult();
+            retVal.GridTariffCollections = new List<GridTariffCollection>();
+            var meteringPointsTariffs = _tariffPriceCache.GetMeteringPointInformation(meteringPoints);
+            foreach (var tariffKey in meteringPointsTariffs.Select(x => x.TariffKey).Distinct())
+            {
+                var gridTariffMeteringPoints = meteringPointsTariffs.Where(x => x.TariffKey == tariffKey).ToList();
+                var gridTariff = await GenerateTariffAndAppendMeteringPoints(tariffKey, paramFromDate, paramToDate, gridTariffMeteringPoints);
+                retVal.GridTariffCollections.Add(gridTariff);
+            }
+            return retVal;
+        }
+
+        public virtual async Task<GridTariffCollection> GenerateTariffAndAppendMeteringPoints(string tariffKey, 
+            DateTimeOffset paramFromDate, 
+            DateTimeOffset paramToDate,
+            List<MeteringPointInformation> meteringPointInformation)
+        {
+            var gridTariff = await QueryTariffAsync(tariffKey, paramFromDate, paramToDate);
+            gridTariff.MeteringPointsAndPriceLevels = new List<MeteringPointsAndPriceLevels>();
+
+            gridTariff = AppendMeteringPointsToPriceLevels(meteringPointInformation, gridTariff);
+            return gridTariff;
+        }
+
+        public virtual GridTariffCollection AppendMeteringPointsToPriceLevels(List<MeteringPointInformation> meteringPointInformation, GridTariffCollection gridTariff)
```

Comment:
Denne add'er til lista inni gridTariff-objektet. Ulogisk å re-assigne da. Enten bør den modifisere objektet som sendes inn, eller returnere en kopi som er endret, og som så må tilordnes. Greit nok at den bare endrer, men da bør den returnere void.

Reply:
Implementerer dette.
Forøvrig er det vel litt filosofisk hvorvidt man foretrekker å endre en inputparameter, eller returnere den.
Ved å returnere den er det utvetydig at funksjonen legger til/endrer data.